### PR TITLE
fix: properly define the subpath ./types in exports

### PR DESCRIPTION
**Pull request by Satont**
Should fix

```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './types' is not defined by "exports"
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,9 @@
             "default": "./out/web.mjs"
         },
         "./types": {
-            "types": "./out/types.d.ts"
+            "types": "./out/types.d.ts",
+            "node": "./out/types.node.js",
+            "default": "./out/types.js"
```
Comment (KnorpelSenf) [inline package.json:46]: In the npm package, `types.js` simply re-exports everything from `types.node.js`. Can we perform the following change and still have it work?
```suggestion
            "node": "./out/types.js",
            "default": "./out/types.js"
```

Comment (KnorpelSenf) [reply package.json:46]: And if yes, then it should also be possible to drop the node-specific export entirely, right?

Comment (KnorpelSenf) [reply package.json:46]: In fact, @kiki-kanri suggested the same thing in https://github.com/grammyjs/hydrate/issues/24#issuecomment-1873433145

Comment (KnorpelSenf) [reply package.json:46]:
```suggestion
            "default": "./out/types.js"
```

**Comment by KnorpelSenf** [comment]
This PQ made me remember what is going on. Thanks!

We should only allow type imports via `/types`. This PQ is not going in the right direction.

Instead, the plugin needs to be fixed. It crashes because it uses `export * from` while it should only use `export type * from`. https://github.com/grammyjs/hydrate/pull/25 Fixes that.

**Review by wojpawlik — APPROVED** [review]

**Pull request merged**